pkg/cloudprovider/providers/vsphere/vsphere_util.go

```diff
@@ -380,30 +380,26 @@ func convertVolPathToDevicePath(ctx context.Context, dc *vclib.Datacenter, volPa
 }
 
 // convertVolPathsToDevicePaths removes cluster or folder path from volPaths and convert to canonicalPath
-func (vs *VSphere) convertVolPathsToDevicePaths(ctx context.Context, nodeVolumes map[k8stypes.NodeName][]string) (map[k8stypes.NodeName][]string, error) {
-	vmVolumes := make(map[k8stypes.NodeName][]string)
-	for nodeName, volPaths := range nodeVolumes {
-		nodeInfo, err := vs.nodeManager.GetNodeInfo(nodeName)
-		if err != nil {
-			return nil, err
-		}
+func (vs *VSphere) convertVolPathsToDevicePaths(ctx context.Context, nodeName k8stypes.NodeName, volPaths []string) ([]string, error) {
```

Comment:
Can we go with the earlier way where we accept nodeVolumes here? ErrVMNotFound error can be handled here.

Reply:
I suppose this change is necessary.
The function before this change received volumes on all nodes, and if any of nodes was not found, the function and VSphere.DisksAreAttached returned error. However attached-disks information needs to be re-registered in DisksAreAttached in case of controller manager restart. That's why I changed this func to handle volumes on a single node so that DisksAreAttached could just ignore removed nodes instead of returning error before reregistration.
Also, DisksAreAttached has a comment which says "// 5b. If VMs are removed from vSphere inventory they are ignored." So I suppose DisksAreAttached shouldn't give error at this point.